test/interface-ipfs-core/ls.spec.js

```diff
@@ -3,36 +3,48 @@
 
 const expect = require('chai').expect
 const isNode = require('detect-node')
+const waterfall = require('async/waterfall')
+const path = require('path')
+
 const FactoryClient = require('../factory/factory-client')
+
 describe('ls', function () {
+  if (!isNode) {
+    return
+  }
+
   let ipfs
   let fc
+  let folderHash
 
   before(function (done) {
     this.timeout(20 * 1000) // slow CI
     fc = new FactoryClient()
-    fc.spawnNode((err, node) => {
-      expect(err).to.not.exist
-      ipfs = node
-      done()
-    })
+    waterfall([
+      (cb) => fc.spawnNode(cb),
+      (node, cb) => {
+        ipfs = node
+        const filesPath = path.join(__dirname, '../data/test-folder')
+        ipfs.util.addFromFs(filesPath, { recursive: true }, cb)
+      },
+      (hashes, cb) => {
+        folderHash = hashes[hashes.length - 1].hash
```

Comment:
Although we should use the value returned to make sure we are fetching the correct one, we should also have the expected hash hard coded, we've captured bugs in the past because the browser was adding different stuff then what it was added in Node.js